crypto/rand/rand_egd.c

```diff
@@ -230,9 +231,7 @@ int RAND_egd_bytes(const char *path, int bytes)
     int num, ret = 0;
 
     num = RAND_query_egd_bytes(path, NULL, bytes);
-    if (num < 1)
-        goto err;
-    if (RAND_status() == 1)
+    if (num < 1 || RAND_status() == 1)
         ret = num;
```

Comment:
No it shouldn't.  if `num >= 1` and `RAND_status() == 1`, then `ret` shouldn't be -1